docs/advanced-features/using-mdx.md

```diff
@@ -0,0 +1,166 @@
+---
+description: Learn how to use @next/mdx in your Next.js project.
+---
+
+# Using MDX with Next.js
+
+Next.js supports MDX through a number of different means, this page will outline some of the ways you can begin integrating MDX into your Next.js project.
+
+## What is MDX?
+
+MDX is a superset of markdown that lets you write JSX directly in your markdown files. It is a powerful way to add dynamic interactivity, and embed components within your content, helping you to bring your pages to life.
+
+### MDX plugins
+
+Internally MDX uses remark and rehype. Remark is a markdown processor powered by a plugins ecosystem. This plugin ecosystem lets you parse code, transform `HTML` elements, change syntax, extract frontmatter, and more.
+
+Rehype is an `HTML` processor, also powered by a plugin ecosystem. Similar to remark, these plugins let you manipulate, sanitize, compile and configure all types of data, elements and content.
+
+To use a plugin from either remark or rehype, you will need to add it to the MDX packages config.
+
+## `@next/mdx`
+
+The `@next/mdx` package is configured in the `next.config.js` file at your projects root. **It sources data from local files**, allowing you to create pages with a `.mdx` extension, directly in your `/pages` directory.
+
+### Setup `@next/mdx` in Next.js
+
+The following steps outline how to setup `@next/mdx` in your Next.js project:
+
+1. Install the required packages:
+
+   ```terminal
+     npm install @next/mdx @mdx-js/loader
+
+     #or
+
+     yarn add @next/mdx @mdx-js/loader
+   ```
+
+2. Require the package and configure to support top level `.mdx` pages. The following adds the `options` object key allowing you to pass in any plugins:
+
+   ```js
+   // next.config.js
+
+   const withMDX = require('@next/mdx')({
+     extension: /\.mdx?$/,
+     options: {
+       remarkPlugins: [],
+       rehypePlugins: [],
+     },
+   })
+   module.exports = withMDX({
+     pageExtensions: ['js', 'jsx', 'md', 'mdx'],
+   })
+   ```
+
+3. Create a new MDX page within the `/pages` directory:
+
+   ```bash
+     - /pages
+       - my-mdx-page.mdx
+     - package.json
+   ```
+
+## Using components, layouts and custom elements
+
+You can now import a React component directly inside your MDX page. Note that `@next/mdx` does **not** support frontmatter (frontmatter is a yaml like key, value pairing that can be used to store data about a page), instead, you can export data from within the `.mdx` file:
```

Comment:
```suggestion
You can now import a React component directly inside your MDX page. Note that `@next/mdx` does **not** support frontmatter (frontmatter is a YAML like key/value pairing that can be used to store data about a page), instead, you can export data from within the `.mdx` file:
```

Should we link to gray-matter as an example of frontmatter?

Reply:
How about this guide? https://mdxjs.com/guides/frontmatter/